src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -42,11 +45,19 @@ public BicepCompilationManager(ILanguageServerFacade server, ICompilationProvide
             this.scheduler = scheduler;
         }
 
-        public void RefreshCompilation(DocumentUri documentUri)
+        public void RefreshCompilation(DocumentUri documentUri, bool reloadBicepConfig = false)
         {
             var compilationContext = this.GetCompilation(documentUri);
+
             if (compilationContext is null)
             {
+                if (reloadBicepConfig &&
+                    workspace.TryGetSourceFile(documentUri.ToUri(), out ISourceFile? sourceFile) &&
+                    sourceFile is BicepFile)
+                {
+                    UpsertCompilationInternal(documentUri, null, sourceFile, reloadBicepConfig);
```

Comment:
>UpsertCompilationInternal [](http://example.com/codeflow?start=20&length=25)

Does this actually recompile anything? I had to create a shallow copy (see lines below) to make it actually recompile. Otherwise, it just reused the existing semantic models.

Reply:
Reverted changes in this file as part of this commit:
https://github.com/Azure/bicep/commit/1ee10711ad7bdd7412967c1ce6ae6f6a1923fa5b

Regarding creating shallow copy, it works without that. Could you please point out a scenario where you think it would not work?

Reply:
Added comment here with more information. Let me know if it makes sense
https://github.com/Azure/bicep/commit/d9739e079ef5badb9363849921c754927e4f24aa

Reply:
Never mind. It makes sense that it worked for you because in my case the compilation existed and I had to force recompile. In your case, it doesn't exist so there's nothing that can be reused.